### PR TITLE
test(core): Fix typing issues in tests (no-changelog)

### PR DESCRIPTION
**Pull request by tomi**
## Summary

Fix typing issues in `cli` tests and enable typechecking for `cli` package as well

## Review / Merge checklist

- [ ] PR title and summary are descriptive. ([conventions](../blob/master/.github/pull_request_title_conventions.md)) 
- [ ] [Docs updated](https://github.com/n8n-io/n8n-docs) or follow-up ticket created.
- [ ] Tests included. 
- [ ] PR Labeled with `release/backport` (if the PR is an urgent fix that needs to be backported)

### CODE REVIEW

**Review by netroy — COMMENTED** [review]
~~should we remove the exception [here](https://github.com/n8n-io/n8n/blob/master/package.json#L16)?~~ nevermind, my brain did not process that there are still error remaining.

File: packages/cli/src/middlewares/listQuery/index.ts [hunk 1]
```diff
@@ -1,8 +1,16 @@
...
+import type { NextFunction, Response } from 'express';
 
-export const listQueryMiddleware = [
+export type ListQueryMiddleware = (
+	req: ListQuery.Request,
+	res: Response,
+	next: NextFunction,
+) => void;
+
+export const listQueryMiddleware: ListQueryMiddleware[] = [
```
Comment (tomi) [inline packages/cli/src/middlewares/listQuery/index.ts:13]: After removing `preserveSymlinks` TS required explicit type here

**Review by netroy — APPROVED** [review]
👏🏽

**Comment by janober** [comment]
Got released with [`n8n@1.53.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.53.0)

**Pull request merged**